client.go

```diff
@@ -71,6 +74,45 @@ type Client struct {
 	cachedEntryLock *sync.RWMutex
 }
 
+// Client is a wrapper around the Resty client
+type HTTPClient struct {
```

Comment:
@ezilber-akamai Any chance we could make this struct private to prevent it from being accessible by outside users? Might also be worth moving this to a separate file for the time being 👍 

Just a bit of context for other reviewers, this is a temporary struct necessary to reduce toe-stepping since we're developing this on the `main` branch

Reply:
Fixed!